### PR TITLE
Feature/charger service

### PR DESCRIPTION
**Pull request by aaronchongth**
Adds a docking service client to the free fleet client which gets triggered when a `MODE_DOCKING` request is received.

### CODE REVIEW

File: free_fleet_client_ros1/src/ClientNode.cpp [hunk 1]
```diff
@@ -283,6 +304,23 @@ bool ClientNode::read_mode_request()
       paused = false;
       emergency = true;
     }
+    else if (mode_request.mode.mode == messages::RobotMode::MODE_CHARGING)
```
Comment (tanyouliang95) [inline free_fleet_client_ros1/src/ClientNode.cpp:307]: Tested with a Kobuki robot by adding a `MODE_DOCKING`. This feature is similarly helpful to trigger a docking sequence with the same `TriggerSrv`.

Comment (aaronchongth) [reply free_fleet_client_ros1/src/ClientNode.cpp:307]: so would you recommend just adding an `or` check? Or does it have it's own service to be used?

Comment (aaronchongth) [reply free_fleet_client_ros1/src/ClientNode.cpp]: https://github.com/open-rmf/free_fleet/pull/74/commits/901e39f806bbf4165938db87d345196cdb3552eb

**Review by tanyouliang95 — APPROVED** [review]
Tested logic on hardware :+1:  Will need to fix the CI, the failing rosdep might be due to `teleop` dep

**Pull request merged**